### PR TITLE
fix: `ActiveSupport` constant conflict in Active Model Serializers instrumentation

### PR DESCRIPTION
**Pull request by flexoid**
When both `ActiveSupport` and `ActiveModelSerializers` instrumentations are loaded,
constant `ActiveSupport::Notifications` inside `ActiveModelSerializers` can be incorrectly
resolved to `OpenTelemetry::Instrumentation::ActiveSupport`.

```
ERROR -- : OpenTelemetry error: Instrumentation: OpenTelemetry::Instrumentation::ActiveModelSerializers unhandled exception during install
...
...
uninitialized constant OpenTelemetry::Instrumentation::ActiveSupport::Notifications
OR
undefined method `subscribe' for Notifications:Module
```

### CODE REVIEW

File: instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def require_dependencies
         end
 
         def register_event_handler
-          ActiveSupport::Notifications.subscribe(event_name) do |_name, start, finish, _id, payload|
+          ::ActiveSupport::Notifications.subscribe(event_name) do |_name, start, finish, _id, payload|
```
Comment (arielvalentin) [inline instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb:38]: Is there a way to write encode this in an automated test case that will reveal this error and avoid regressions?

Comment (flexoid) [reply instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb:38]: Well, I thought about it. But it is reproduced only when both `opentelemetry-instrumentation-active_support` and `opentelemetry-instrumentation-active_model_serializers` are included to the app, and these gems have isolated test specs. So it's a bit hard to test the conflict.

Comment (arielvalentin) [reply instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb:38]: I wonder if this is something we may be able to detect in the `all` test suite? cc: @open-telemetry/ruby-approvers

Comment (flexoid) [reply instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb:38]: Done! Reproduced with a quite simple test case. Thanks for the suggestion 🙂

Comment (arielvalentin) [reply instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb:38]: Sweet! Did the tests fail when you removed the top level qualifier `::`?

**Review by fbogsany — APPROVED** [review]

File: instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb (hunk 1 above)
Comment (flexoid) [reply instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb:38]: Yep, that's what I started from.

Comment (flexoid) [reply instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb:38]:
```
Finished in 0.006173s, 161.9958 runs/s, 0.0000 assertions/s.

  1) Error:
OpenTelemetry::Instrumentation::All::ActiveModelSerializers instrumentation#test_0001_does not conflict with ActiveSupport instrumentation:
NameError: uninitialized constant OpenTelemetry::Instrumentation::ActiveSupport::Notifications
```

**Pull request merged**